src/components/base-input/index.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { memo } from "react";
+
+import { type BaseInputData } from "./base-input.types";
+
+const inputTypeDefaultClasses = {
+  textarea: "textarea-bordered textarea h-20",
+  input: "input input-bordered"
+};
+
+// TODO(aliabudiab112@gmail.com): Add a "...restProps" parameter to handle properties for both input and textarea components.
+
+const BaseInput: React.FC<BaseInputData> = ({
+  label = "",
+  placeholder = "",
+  inputType = "textarea",
+  containerCustomClasses = "",
+  labelCustomClasses = "",
+  inputCustomClasses = ""
```

Comment:
It seems the onChange handler is missing.

Reply:
No, I intentionally left it for us to address when we start handling logical actions, as I mentioned in my TODO comment. However, if you prefer, I can add it now